language/move-vm/runtime/src/loader.rs

```diff
@@ -65,17 +65,35 @@ where
         }
     }
 
-    fn insert(&mut self, key: K, binary: V) -> &Arc<V> {
+    fn insert(&mut self, key: K, binary: V) -> PartialVMResult<&Arc<V>> {
+        let idx = self.binaries.len();
+        if self.id_map.insert(key, idx).is_some() {
+            return Err(PartialVMError::new(StatusCode::LINKER_ERROR)
```

Comment:
If it's in internal implementation issue that would cause the error then I'd go for the `UNKNOWN_INVARIANT_VIOLATION` (or even add a `LINKER_INVARIANT_VIOLATION` status).

Reply:
I'll go for unknown for now -- I vaguely recall there being bytecode format implications to adding a new status code?  So don't want to get tripped up on that at the moment.